### DISCUSSION TITLE
Add something like fast forward [with code example]

### DISCUSSION THREAD

**Discussion by Smallinger** [Zibbp/ganymede#169] [Ideas]
Hey is it possible to add something like youtube fast forwards left and right arrow?
with like a small notification in the player on right side for +5 seconds and on left side for -5 seconds.
![chrome_MKzaquuVRf](https://user-images.githubusercontent.com/8080568/231356299-c06b0031-b44b-4f8e-99a8-1ae38f1b2272.png)

i test this on the browser console in ganymede.
```
const video=document.getElementById("#vjs_video_3");
forward=()=>{
  skip(5);
}
backward=()=>{
   skip(-5);
}
skip=(time)=>{
  video.currentTime=video.currentTime+time;
}
(time)=>{
  video.currentTime=video.currentTime+time;
}
```
```
document.addEventListener("keydown",(e)=>{
    if(e.keyCode==37){       //left arrow
        backward()
    }else if(e.keyCode==39){ //right arrow
        forward()
    }
  }
)
```

or with a plugin for video.js ?
https://github.com/mister-ben/videojs-seek-buttons

**Comment by Zibbp** [comment]
Arrow key seeking does exist right now but the player needs to be focused (click anywhere in the player) then you can use the functionality. I can extend the functionality to listen to keys on the entire page, like the code you suggested. Creating a notification on the player when seeking can be done as well.

**Reply by Smallinger** [reply]
Yeah i think thats would be cool.
I don't know if VideoJS supports it but later you might add something like.

**1. Preview thumbnail in progress-slider.**

![firefox_rE9bemq4yf](https://user-images.githubusercontent.com/8080568/232308299-fc0331ca-dd32-4ccc-938d-2c22747a3c5b.png)

**2. Maybe something like this, I don't know how Youtube calls it.**

 ![firefox_7fr0YWcwOH](https://user-images.githubusercontent.com/8080568/232308177-d4adf42b-96ac-4f4c-acbc-2e94927a6c57.gif)

**Reply by Zibbp** [reply]
I know VideoJS has a plugin for scrub thumbnails like your first image. The second image, I haven't seen anywhere in an open source player yet. I've wanted to do srub thumbnails but never got around to it because each video needs to be processed and generate all of those thumbnails files which need to be stored.

I've also been keeping my on eye on [vidstack](https://github.com/vidstack/player) as a VideoJS replacement. Working with VideoJS, I'm not a huge fan. Vidstack is also the successor of [Plyr](https://github.com/sampotts/plyr), which was used previously, and I thoroughly enjoyed.

I'll revisit the scrub thumbnails when I implement Vidstack.

**Reply by Zibbp** [reply]
Release [v1.2.8](https://github.com/Zibbp/ganymede-frontend/releases/tag/v1.2.8) allows seeking without focusing the player (listens to the document). I did not add the on-screen "seeking 20 seconds" popup as I don't want to invest more time into creating custom VideoJS functionality. I plan on moving over to Vidstack as soon as their community skin is released, which is supposed to be the next release.
